alert-common/src/main/java/com/synopsys/integration/alert/common/action/CustomEndpointManager.java

```diff
@@ -24,37 +24,45 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.BiFunction;
 import java.util.function.Function;
 
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
 
 import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.rest.HttpServletContentWrapper;
 import com.synopsys.integration.alert.common.rest.model.FieldModel;
 
 @Component
 public class CustomEndpointManager {
     public static final String CUSTOM_ENDPOINT_URL = "/api/function";
-    private Map<String, Function<FieldModel, ResponseEntity<String>>> endpointFunctions = new HashMap<>();
+    private final Map<String, BiFunction<FieldModel, HttpServletContentWrapper, ResponseEntity<String>>> endpointFunctions = new HashMap<>();
 
     public boolean containsFunction(String functionKey) {
         return endpointFunctions.containsKey(functionKey);
     }
 
     public void registerFunction(String functionKey, Function<FieldModel, ResponseEntity<String>> endpointFunction) throws AlertException {
+        if (containsFunction(functionKey)) {
+            throw new AlertException("A custom endpoint is already registered for " + functionKey);
+        }
```

Comment:
This part is duplicated between methods. Can we call the method below here?

Reply:
Ugh, meant to do this on line 51.